evennia/contrib/turnbattle/tb_items.py

```diff
@@ -0,0 +1,1397 @@
+                            "|wAttack:|n Attack a target, attempting to deal damage.|/" +
+                            "|wPass:|n Pass your turn without further action.|/" +
+                            "|wDisengage:|n End your turn and attempt to end combat.|/" +
+                            "|wUse:|n Use an item you're carrying.")
+        else:
+            super(CmdCombatHelp, self).func()  # Call the default help command
+
+
+class CmdUse(MuxCommand):
+    """
+    Use an item.
+
+    Usage:
+      use <item> [= target]
+
+    An item can have various function - looking at the item may
+    provide information as to its effects. Some items can be used
+    to attack others, and as such can only be used in combat.
+    """
+
+    key = "use"
+    help_category = "combat"
+
+    def func(self):
+        """
+        This performs the actual command.
+        """
+        # Search for item
+        item = self.caller.search(self.lhs, candidates=self.caller.contents)
+        if not item:
+            return
+        
+        # Search for target, if any is given
+        target = None
+        if self.rhs:
+            target = self.caller.search(self.rhs)
+            if not target:
+                return
+        
+        # If in combat, can only use items on your turn
+        if is_in_combat(self.caller):
+            if not is_turn(self.caller):
+                self.caller.msg("You can only use items on your turn.")
+                return
+            
+        if not item.db.item_func: # Object has no item_func, not usable
+            self.caller.msg("'%s' is not a usable item." % item.key.capitalize())
+            return
+            
+        if item.attributes.has("item_uses"): # Item has limited uses
+            if item.db.item_uses <= 0: # Limited uses are spent
+                self.caller.msg("'%s' has no uses remaining." % item.key.capitalize())
+                return
+        
+        # If everything checks out, call the use_item function
+        use_item(self.caller, item, target)
+
+
+class BattleCmdSet(default_cmds.CharacterCmdSet):
+    """
+    This command set includes all the commmands used in the battle system.
+    """
+    key = "DefaultCharacter"
+
+    def at_cmdset_creation(self):
+        """
+        Populates the cmdset
+        """
+        self.add(CmdFight())
+        self.add(CmdAttack())
+        self.add(CmdRest())
+        self.add(CmdPass())
+        self.add(CmdDisengage())
+        self.add(CmdCombatHelp())
+        self.add(CmdUse())
+        
+"""
+----------------------------------------------------------------------------
+ITEM FUNCTIONS START HERE
+----------------------------------------------------------------------------
+
+These functions carry out the action of using an item - every item should
+contain a db entry "item_func", with its value being a string that is
+matched to one of these functions in the ITEMFUNCS dictionary below.
```

Comment:
While this way of mapping is fine, a more Evennia-esque way would be to have these functions in a separate module and use the `callables_from_module` Evennia utility - it returns a dictionary keyed by their `def` names in the module, with values being the callables. That way you could easily have people add new stuff by just having them add the path to a new module with item-funcs to some setting (see for example how lockfuncs, inlinefuncs etc work). Not necessary but probably more consistent with how other similar function-plugins work in Evennia.

Reply:
Seems good! I'll keep that in mind - I'm interested in going through all the turnbattle modules and kind of cleaning them up a bit at some point.